File: py/prob_49.py
class Solution:
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        anagram_groups = {}
        for string in strs:
            string_sorted = "".join(sorted(string))
            if string_sorted not in anagram_groups.keys():
                anagram_groups[string_sorted] = [string]
            else:
                anagram_groups[string_sorted].append(string)
        return list(anagram_groups.values())

File: js/prob_49.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
const groupAnagrams = strs => {
     let anagram_groups = {}
     for (let string of strs) {
        let string_sorted = string.split("").sort().join()
        
        if (!(string_sorted in anagram_groups)) {
            anagram_groups[string_sorted] = [string]
        }
        else {
            anagram_groups[string_sorted].push(string)
        }
     }
                
    return Object.values(anagram_groups)
}